### PR TITLE
7-add-Test-to-FlagBrowser

### PR DESCRIPTION
**Pull request by CafeKrem**
add test for FBByScopePagePresenter

### CODE REVIEW

**Comment by StevenCostiou** [comment]
This is more than one year old.
Is it still on the table?

**Comment by CafeKrem** [comment]
I don't know Cyrille Ask to add test to his flag browser.Monday I will test on pharo 9 ans improve it if it's needed

File: src/NewTools-FlagBrowser-Tests/FBByFlagsPagePresenterTest.class.st [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	#category : #'NewTools-FlagBrowser-Tests-Presenters'
+}
+
+{ #category : #running }
+FBByFlagsPagePresenterTest >> instanciateFlagPage [
+	^ FBByFlagsPagePresenter new
+]
+
+{ #category : #tests }
+FBByFlagsPagePresenterTest >> selectFlagaRandomFalgForTest [
```
Comment (CafeKrem) [inline src/NewTools-FlagBrowser-Tests/FBByFlagsPagePresenterTest.class.st:16]: Add parameter

File: src/NewTools-FlagBrowser-Tests/FBClassMock.class.st [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+{ #category : #compiling }
+FBClassMock >> compile: test [
+	isCompileCall := true
+]
+
+{ #category : #initialization }
+FBClassMock >> initialize [
+	isCompileCall := true
```
Comment (CafeKrem) [inline src/NewTools-FlagBrowser-Tests/FBClassMock.class.st:17]: Should bé false

File: src/NewTools-FlagBrowser-Tests/FBFlagPresentationPresenterTest.class.st [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+{ #category : #running }
+FBFlagPresentationPresenterTest >> setUp [
+	"Hooks that subclasses may override to define the fixture of test."
+
+	flagPage := self instanciateFlagPage.
+	flagPage locationsTable
+		items:
+			{self getLocationFlagMethod.
+			self getLocationFlagMethod2}
```
Comment (CafeKrem) [inline src/NewTools-FlagBrowser-Tests/FBFlagPresentationPresenterTest.class.st:41]: Maybe inlined or use a factory

File: src/NewTools-FlagBrowser-Tests/FBMock.class.st [hunk 4]
```diff
@@ -15,3 +15,28 @@ FBMock >> = anObject [
...
 ]
+
+{ #category : #running }
+FBMock >> flagMethod [
+	self flag: #aRandomFlagForTest
+]
+
+{ #category : #running }
+FBMock >> flagMethod2 [
+	self flag: #aRandomFlagForTest
```
Comment (CafeKrem) [inline src/NewTools-FlagBrowser-Tests/FBMock.class.st:26]: Same flag

**Comment by StevenCostiou** [comment]
Hi Clément, thanks.
I see the flag browser is not loaded by default, neither in Pharo nor in NewTools.
Is it supposed to be part of NewTools? What is the status of that project? (I wonder if it loads and works)

**Comment by CafeKrem** [comment]
Cyril asked me to push those codes on newTools so I think he wants to integrate it. 
"What is the status of that project?"
you should ask @jecisc  for more information

**Comment by StevenCostiou** [comment]
Ok :) Thanks @CafeKrem :)

**Pull request merged**